### PR TITLE
Release 3.2.0

### PR DESCRIPTION
**Pull request by lucas-koehler**
- Update architecture diagrams
- add release news entry
- fix broken links
- remove vue 2
- updated api docs

### CODE REVIEW

**Comment by lucas-koehler** [comment]
When reviewing, please pay special attention to the suggested news entry. Also consider that the news date is embedded in the file name (currently 24.01.). We might need to update this depending on the concrete release day :D

File: content/docs/integrations/vue.mdx [hunk 1]
```diff
@@ -11,11 +11,10 @@ Currently this integration is in a preview state!
 :::
 
 With version 2.5 of JSON Forms we added support for Vue 2 (`@jsonforms/vue2`) and Vue 3 (`@jsonforms/vue`).
-Support for Vue 2 will be phased out once the community moves on.
+Support for Vue 2 was dropped with JSON Forms 3.2.
+The last official release containg Vue 2 is [v3.1.0](https://github.com/eclipsesource/jsonforms/tree/v3.1.0).
 
-For Vue we provide a HTML5 based renderer set, `@jsonforms/vue-vanilla` for Vue 3 and `@jsonforms/vue2-vanilla` for Vue 2.
+For Vue we provide a HTML5 based renderer set, `@jsonforms/vue-vanilla` for Vue 3.
```
Comment (sdirix) [inline content/docs/integrations/vue.mdx:17]: We should also mention vue vuetify which is in a preview state

File: content/docs/renderer-sets.mdx [hunk 2]
```diff
@@ -31,12 +31,12 @@ The Angular Material UI Renderer Set is based on [Angular Material](https://mate
...
 
 The Vue Renderer Set is based on plain HTML with a set of custom styles.
-For more information, see the [Vue Vanilla Readme](https://github.com/eclipsesource/jsonforms/blob/master/packages/vue/vue-vanilla/README.md).
+For more information, see the [Vue Vanilla Readme](https://github.com/eclipsesource/jsonforms/blob/master/packages/vue-vanilla/README.md).
 
-## Vue2 Vuetify Renderer Set
+## Vue Vuetify Renderer Set
 
-The Vue2 Vuetify Set provides a Vuetify based renderer set for JSON Forms Vue 2.
+The Vue Vuetify Set provides a Vuetify based renderer set for JSON Forms.
```
Comment (sdirix) [inline content/docs/renderer-sets.mdx:38]: We should explain that Vue Vuetify is in a preview state and has some known bugs. We are looking for contributors to fix them. We can use a "Note" for this.

**Review by sdirix — CHANGES_REQUESTED** [review]
Looks good!

We should add the following changes:
 - Adapt the news on the landing page too (it's hard coded there atm)
 - In the news link to the changelog in the repository
 - Update the used JSON Forms version of the website to 3.2.0

**Comment by lucas-koehler** [comment]
@sdirix thanks for the review, I adapted the mentioned points.

**Comment by lucas-koehler** [comment]
@sdirix I updated the renderer set feature table and removed some more vue 2 references i found. Hopefully, everything is cleaned up now :)

**Comment by sdirix** [comment]
@lucas-koehler I just saw that we seemingly have a regression? For some reason some of the inputs in the landing page form are using an outlined input

File: content/news/2024-01-24-jsonforms-3-2-0.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+author_url: https://twitter.com/jsonforms
+author_image_url: /img/logo.svg
+---
+We released v3.2.0 🎉. Besides various fixes and improved developer tooling, this release adds middleware support to all bindings. This makes it a breeze to hook into data updates!
+
+With this release, the Angular core and material renderer set finally are up to date again supporting Angular 16 and 17. Special thanks to [JBBianchi](https://github.com/JBBianchi) for contributing this massive upgrade.
+
+You can find the full changelog in the [Github release](https://github.com/eclipsesource/jsonforms/releases/tag/v3.2.0).
+
+We welcome any feedback of curious users. Any questions? Check out our [community forum](http://jsonforms.discourse.group).
```
Comment (sdirix) [inline content/news/2024-01-24-jsonforms-3-2-0.md]:
```suggestion
We released v3.2.0 🎉. Besides various fixes and improved developer tooling, this release adds middleware support to all bindings. This makes it a breeze to hook into data updates!

With this release, the Angular bindings and Angular Material renderer set now support Angular 16 and 17. Special thanks to [JBBianchi](https://github.com/JBBianchi) for contributing this massive upgrade.

You can find the full changelog in the [Github release](https://github.com/eclipsesource/jsonforms/releases/tag/v3.2.0).

We welcome any feedback of curious users. Any questions? Check out our [community forum](http://jsonforms.discourse.group).
```

File: content/pages/index.js [hunk 4]
```diff
@@ -107,11 +107,15 @@ function Home() {
           <section className={styles.sectionNews}>
             <Card className={styles.newsCard}>
               <CardContent>
-                Great news! We are in the final stretches of the long awaited JSON Forms 3.0 release. Today we published the release candidate "3.0.0-rc.0" 🎉
+                We released v3.2.0 🎉. Besides various fixes and improved developer tooling, this release adds middleware support to all bindings. This makes it a breeze to hook into data updates!
                 <br/>
-                We welcome any feedback of curious users. Any questions? Check out our community forum <a href="https://jsonforms.discourse.group">jsonforms.discourse.group</a>.
+                With this release, the Angular core and material renderer set finally are up to date again supporting Angular 16 and 17. Special thanks to <a href="https://github.com/JBBianchi">JBBianchi</a> for contributing this massive upgrade.
```
Comment (sdirix) [inline content/pages/index.js]:
```suggestion
                With this release, the Angular bindings and Angular Material renderer set now support Angular 16 and 17. Special thanks to <a href="https://github.com/JBBianchi">JBBianchi</a> for contributing this massive upgrade.
```

File: content/pages/index.js [hunk 5]
```diff
@@ -107,11 +107,15 @@ function Home() {
...
+                <br />
+                You can find the full changelog in the <a href="https://github.com/eclipsesource/jsonforms/releases/tag/v3.2.0">Github release</a>.
+                <br />
+                We welcome any feedback of curious users. Any questions? Check out our <a href="http://jsonforms.discourse.group">community forum</a>.
```
Comment (sdirix) [inline content/pages/index.js]:
```suggestion
                We welcome any feedback of curious users. Any questions? Check out our <a href="http://jsonforms.discourse.group">community forum</a>. See also our <a href="https://github.com/eclipsesource/jsonforms/blob/master/MIGRATION.md">migration guide</a>.
```

**Review by sdirix — APPROVED** [review]
Looks good to me, only minor comments

**Pull request merged**